tests/test_utils/test_misc.py

```
@@ -571,6 +572,16 @@ def test_nothing(self):
             warn_or_raise(msg, behavior=None)
 
 
+class TestSuppressWarnings:
+    """Tests for the suppress_warnings context manager."""
+
+    def test_message_filter_applies_action(self):
+        """A message pattern applies the action to matching warnings."""
+        with suppress_warnings(message="boom", action="error"):
+            with pytest.raises(UserWarning, match="boom"):
+                warnings.warn("boom", UserWarning)
```

Ruff (0.15.21)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)